[EdgeTpu] ConfigSetting

imsw0529 opened the pull request:
- move parse function to ConfigSetting
- configType and getConfigSetting
- expand Cfg type for .edgetpucfg
- updateBaseModelField separate logic

ONE-vscode-DCO-1.0-Signed-off-by: Seongwon Im <imsw0529@gmail.com>

imsw0529 committed:
[EdgeTpu] Move parse function to ConfigSetting

- move parse function to ConfigSetting.ts
- change ConfigObject's member configSetting to configType

ONE-vscode-DCO-1.0-Signed-off-by: Seongwon Im <imsw0529@gmail.com>

imsw0529 committed:
[EdgeTpu] Expand Cfg type for .edgetpucfg

- define OneCfg, EdgeTpuCfg type to ConfigSettings
- expand Cfg type
- make getter for ConfigSetting

ONE-vscode-DCO-1.0-Signed-off-by: Seongwon Im <imsw0529@gmail.com>

imsw0529 committed:
[EdgeTpu] Select updateBaseModelField logic based on ConfigSetting

- select sections by ConfigSetting
- add updateOutPath for edgetpucfg

ONE-vscode-DCO-1.0-Signed-off-by: Seongwon Im <imsw0529@gmail.com>

SeungHyunH commented on src/OneExplorer/ConfigObject.ts:
```diff
@@ -25,21 +25,26 @@ import { Logger } from "../Utils/Logger";
+// type CfgOneImportTf = any;
+
+type CfgType = {
+  default: any;
```
I think this type's naming is "one" better than  "default" because "default" is ambiguous.

SeungHyunH commented on src/OneExplorer/ConfigObject.ts:
```diff
@@ -122,21 +126,33 @@ export class ConfigObj {
+   * @brief getter for config setting
+   */
+  public get getConfigSetting(): ConfigSetting {
+    let configSetting: ConfigSetting = new OneConfigSetting();
```
this code is must be update to switch-case because ConfigSetting is created twice

SeungHyunH replied:
Please refer to the "NodeFactory" code in "OneExplorer".

imsw0529 committed:
[EdgeTpu] Fix CfgType key name

- change "default" to "one"

ONE-vscode-DCO-1.0-Signed-off-by: Seongwon Im <imsw0529@gmail.com>

SeungHyunH commented on src/OneExplorer/ConfigSettings/EdgeTpuConfigSetting.ts:
```diff
@@ -19,16 +19,35 @@ import * as vscode from "vscode";
+    kSection: string
+  ): void {
+    const ext = ".tflite";
+    let outpath = newpath.slice(0, -ext.length);
```
I think this code can be migration by using path library

imsw0529 committed:
[EdgeTpu] Change getConfigSetting logic to switch

- change getConfigSetting's branch logic to switch
- change updateOutPath logic to using path module

ONE-vscode-DCO-1.0-Signed-off-by: Seongwon Im <imsw0529@gmail.com>

imsw0529 commented:
I fixed that you reviewed.
